lib/Core/StatsTracker.cpp

```diff
@@ -243,12 +241,13 @@ StatsTracker::StatsTracker(Executor &_executor, std::string _objectFilename,
 
   if (OutputStats) {
     statsFile = executor.interpreterHandler->openOutputFile("run.stats");
-    assert(statsFile && "unable to open statistics trace file");
-    writeStatsHeader();
-    writeStatsLine();
+    if (statsFile) {
+      writeStatsHeader();
+      writeStatsLine();
 
-    if (StatsWriteInterval > 0)
-      executor.addTimer(new WriteStatsTimer(this), StatsWriteInterval);
+      if (StatsWriteInterval > 0)
+        executor.addTimer(new WriteStatsTimer(this), StatsWriteInterval);
+    }
```

Comment:
The logic here is that you need to set it to 0 to disable it (in case you want to use `-stats-write-after-instructions`), as otherwise the following issues an error message:
https://github.com/klee/klee/blob/224532d66198c80c89b6017e14de4bbed53e305e/lib/Core/StatsTracker.cpp#L187-L190
Values other than 0 are not possible (in the else branch), as `StatsWriteInterval` is `unsigned`.

Reply:
That's not what I meant (also this will probably change). I meant you should handle the error case for `if (statsFile)` and `if (istatsFile)`.

Reply:
Oh, I see. Sorry, will change that.